### PR TITLE
docs(design): add docs for theming `@daffodil/design`.

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
Documents how to use the out-of-the-box daffodil theme features.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/design/docs/theming.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Theming
+
+The **daff-theme** utility library is for theming the daffodil
+design library. It contains many useful mixins, functions, 
+and variables.
+
```
Comment (xelaint) [inline libs/design/docs/theming.md]: extra empty line

File: libs/design/docs/theming.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Step 1: Add the path to your angular configuration file.
+Within your project's `angular.json` append the following to the key `projects.project-name.architect.build.options`
+
+```json
+"build":{
+  "options":{
+    "stylePreprocessorOptions": {
+              "includePaths": [
```
Comment (xelaint) [inline libs/design/docs/theming.md]: too many tabs

File: libs/design/docs/theming.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                "./node_modules/@daffodil/design"
+              ]
+      }
+  }
+}
+```
+## Step 2: Import the style sheets inside of your projects global styling sheet.
```
Comment (xelaint) [inline libs/design/docs/theming.md]: `project's`

File: libs/design/docs/theming.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+In your project's global style sheet( typically `styles.scss` in the project's root directory; e.g.`src/` or `projects/my-project/src` ) add the following code and then all of Daffodil's styling tools will be available.
```
Comment (xelaint) [inline libs/design/docs/theming.md]: ```In your project's global style sheet (typically `styles.scss` in the project's root directory; e.g.`src/` or `projects/my-project/src`), add the following code for all of Daffodil's styling tools to be available.```

File: libs/design/docs/theming.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Theming
+
+The **daff-theme** utility library is for theming the daffodil
+design library. It contains many useful mixins, functions, 
+and variables.
+
+
+## Step 1: Add the path to your angular configuration file.
+Within your project's `angular.json` append the following to the key `projects.project-name.architect.build.options`
```
Comment (xelaint) [inline libs/design/docs/theming.md]: comma before `append`?

File: libs/design/docs/theming.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
+# Theming
+
+The **daff-theme** utility library is for theming the daffodil
+design library. It contains many useful mixins, functions, 
+and variables.
```
Comment (lderrickable) [inline libs/design/docs/theming.md]: This sounds awkward to me. What do you think about "The **daff-theme** utility library consists of mixins, functions, and variables for theming the daffodil design library."?

File: libs/design/docs/theming.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+## Step 1: Add the path to your angular configuration file.
+Within your project's `angular.json`, append the following to the key `projects.project-name.architect.build.options`
```
Comment (lderrickable) [inline libs/design/docs/theming.md]: should be 
"...to the `projects.project-name.architect.build.options` key:" 
or 
"...to the key, `projects.project-name.architect.build.options`:"

File: libs/design/docs/theming.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  "options":{
+    "stylePreprocessorOptions": {
+        "includePaths": [
+            "./node_modules/@daffodil/design"
+          ]
+      }
+  }
+}
+```
+## Step 2: Import the style sheets inside of your project's global styling sheet.
```
Comment (lderrickable) [inline libs/design/docs/theming.md]: Should be: "Import the style sheets into your project's global styling sheet."

File: libs/design/docs/theming.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+In your project's global style sheet (typically `styles.scss` in the project's root directory; e.g.`src/` or `projects/my-project/src`), add the following code for all of Daffodil's styling tools to be available.
```
Comment (lderrickable) [inline libs/design/docs/theming.md]: To make all of Daffodil's styling tools available, add the following code to your project's global style sheet (typically `styles.scss` in the project's root directory; e.g.`src/` or `projects/my-project/src`):

File: libs/design/docs/theming.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+## Theming Installation
+
+If you’d like to find out more about how to use the library and how to take advantage of its capabilities, see the [theming overview](../theming.md).
+
+
+> This setup assumes that you’ve already installed the `@daffodil/design` package.
+
+**Add the package path to your angular configuration file**
+Within your project's `angular.json` file, include the path to the `@daffodil/design` package inside of  `stylePreprocessorOptions``.includePaths`:
```
[the remark this answers is not in the document]
Comment (Nolan-Arnold) [reply libs/design/docs/theming.md]: Okay, these two typos are fixed now.

File: libs/design/guides/theming/installation.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
+# Daffodil Theming Installation Guide
+
+## Theming Installation
+
+If you’d like to find out more about how to use the library and how to take advantage of its capabilities, see the [theming overview](../theming.md).
+
```
Comment (xelaint) [inline libs/design/guides/theming/installation.md]: remove extra empty line

File: libs/design/guides/theming/installation.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+> This setup assumes that you’ve already installed the `@daffodil/design` package.
+
+**Add the package path to your angular configuration file**
+Within your project's `angular.json` file, include the path to the `@daffodil/design` package inside of  `stylePreprocessorOptions.includePaths`:
+
```
Comment (xelaint) [inline libs/design/guides/theming/installation.md]: remove extra empty line

File: libs/design/guides/theming/installation.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+    json
```
Comment (xelaint) [inline libs/design/guides/theming/installation.md]: this block of code needs to be wrapped in code format (```)

File: libs/design/guides/theming/installation.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+          }
+      }
+    }
+    ...
+
+**Import the SASS stylesheets in****to** **your project's global stylesheet**
+To make all of Daffodil's SASS styling utilities easily available, add the following code to your project's global stylesheet (typically `src/styles.scss`):
+
+
+    scss
```
Comment (xelaint) [inline libs/design/guides/theming/installation.md]: this block of code needs to be wrapped in code format (```)

File: libs/design/guides/theming/installation.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+        "stylePreprocessorOptions": {
+            "includePaths": [
+                "./node_modules/@daffodil/design"
+              ]
+          }
+      }
+    }
+    ...
+
+**Import the SASS stylesheets in****to** **your project's global stylesheet**
```
Comment (xelaint) [inline libs/design/guides/theming/installation.md]: `in****to** **` need to fix this

File: libs/design/guides/theming/installation.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+To make all of Daffodil's SASS styling utilities easily available, add the following code to your project's global stylesheet (typically `src/styles.scss`):
+
+
+    scss
+    
+    @import 'daff-theme';
+
```
Comment (xelaint) [inline libs/design/guides/theming/installation.md]: remove extra empty line

File: libs/design/guides/theming/installation.md [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+                "./node_modules/@daffodil/design"
+              ]
+          }
+      }
+    }
+    ...
+
+**Import the SASS stylesheets in****to** **your project's global stylesheet**
+To make all of Daffodil's SASS styling utilities easily available, add the following code to your project's global stylesheet (typically `src/styles.scss`):
+
```
Comment (xelaint) [inline libs/design/guides/theming/installation.md]: remove extra empty line

File: libs/design/guides/theming/installation.md [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+    scss
+    
+    @import 'daff-theme';
+
+
+## Configure Your Theme
+
+Configure the global theme for the Daffodil components that will be used in your project.
+
```
Comment (xelaint) [inline libs/design/guides/theming/installation.md]: remove extra empty line

File: libs/design/guides/theming/installation.md [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+
+    scss
```
Comment (xelaint) [inline libs/design/guides/theming/installation.md]: this block of code needs to be wrapped in code format (```)

File: libs/design/guides/theming/installation.md [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+    @import 'daff-theme';
+    ...
+    $primary: daff-configure-palette($daff-yellow, 60);
+    $secondary: daff-configure-palette($daff-purple, 60);
+    $tertiary: daff-configure-palette($daff-blue, 60);
+    $theme: daff-configure-theme($primary, $secondary, $tertiary, 'light');
+    ...
+    @include daff-theme($theme);
+
```
Comment (xelaint) [inline libs/design/guides/theming/installation.md]: remove extra empty line

File: libs/design/guides/theming/installation.md [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+
+> Note: This does not add the sass utilities into your individual component files. Those files are compiled by the Angular CLI separately, and currently need their own imports if you are planning on using the `daff-theme`  features within your component scss files.
+
+
```
Comment (xelaint) [inline libs/design/guides/theming/installation.md]: remove extra empty line

Comment (Nolan-Arnold) [reply libs/design/guides/theming/installation.md]: Okay these should all be resolved in the latest commit.

**Comment by lderrickable** [comment]
lgtm

**Pull request merged**